gcloud/datastore/__init__.py

```diff
@@ -69,6 +69,50 @@
 _DATASET_ENV_VAR_NAME = 'GCLOUD_DATASET_ID'
 
 
+def _find_true_dataset_id(dataset_id, connection=None):
```

Comment:
ISTM that making people know about `s-foo` is an anti-goal, even in the short term.